Added ability to get the id of email attachments

craftfortress opened the pull request:
That's pretty much it.

Commit:
Added getMessageAttachmentId method so we can get the id of email attachments

Commit:
Formatting improved ever so slightly

jasonjoh commented:
Thanks @craftfortress! I'll review this ASAP

jasonjoh commented on mail-api.js:
```diff
@@ -96,6 +96,34 @@ module.exports = {
   },
   
+
+  getMessageAttachmentId: function(parameters, callback) {
```
This function doesn't return the attachment ID, it returns a collection of attachments. A better name for this function would be `getMessageAttachments`.

jasonjoh commented on mail-api.js:
```diff
@@ -96,6 +96,34 @@ module.exports = {
+      user: parameters.user
+    };
+
+    base.makeApiCall(apiOptions, function(error, response) {
```
You should support optional query parameters so callers can do things like `$select`. You can follow the pattern in `getMessages` for example:

```
if (parameters.odataParams !== undefined) {
    apiOptions['query'] = parameters.odataParams;
}
```

jasonjoh commented on mail-api.js:
```diff
@@ -96,6 +96,34 @@ module.exports = {
     });
   },
   
+
```
We should include documentation comments so this can get added into the reference.

jasonjoh reviewed (changes requested):
@craftfortress I added some review items that should be addressed before this gets merged. I figured I'd give you first shot at them, but if you can't get to it I can make the requested changes.

craftfortress commented:
Thanks I will try and look into this over the next week

jasonjoh committed:
Added docs and tweaked function

- Renamed function to `getMessageAttachments` to better reflect what it does.
- Added support fo OData query params
- Added JSDOC comments

These should take care of the requested changes on https://github.com/jasonjoh/node-outlook/pull/27.

craftfortress commented:
Sorry I never got around to this, it was on my mind
